File: 202-happy-number/happy-number.java
class Solution {
    public boolean isHappy(int n) {
        int fast = n;
        int slow = n;

        do{
          slow = square(slow);
          fast = square(square(fast));
        } while(slow != fast);

        return slow == 1;
    }
    private int square(int n){

      int sum = 0;
      while(n != 0){
        int mod = n%10;
        sum += mod*mod;
        n = n/10;
      }
      return sum;
    }
}

File: 202-happy-number/happy-number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        slow = fast = n

        def square(num):
            tot = 0

            while num > 0:
                dig = num%10
                tot += (dig*dig)
                num = num//10

            return tot

        while True:            
            slow = square(slow)
            fast = square(square(fast))
            if slow == fast:
                break
        
        return slow == 1 or fast == 1

